geopandas/base.py

```diff
@@ -41,7 +41,11 @@ def _delegate_binary_method(op, this, other, *args, **kwargs):
     # type: (str, GeoSeries, GeoSeries) -> GeoSeries/Series
     this = this.geometry
     if isinstance(other, GeoPandasBase):
-        this, other = this.align(other.geometry)
+        if not this.index.equals(other.index):
+            warn('The index of the two GeoSeries are different')
```

Comment:
```suggestion
            warn('The indices of the two GeoSeries are different.')
```
Minor language-based suggestion.

Reply:
Thanks @martinfleis 

I've updated the text to use "indices" and the trailing dot.